<a href="https://colab.research.google.com/github/NirmalVignu/ineuron_MLDL_assignments/blob/main/Assignment_stats_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import scipy.stats as sts
from scipy.stats import norm
import math
import numpy as np

Problem Statement 1:

Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.

In [2]:
p_mean = 100
p_std = 15
n = 36
sample_mean = 108
alpha = 0.05

SE = p_std/n**0.5
print(f"SE: {SE}")
Z = (sample_mean-p_mean)/SE
print(f"Z_score: {Z}")

SE: 2.5
Z_score: 3.2


by looking at z- table and p-value associated with 3.20 is 0.9993 
The probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.

Step-4: Sice the probability of having mean glucose level more than or equals to 108  is 0.0007 which is  less than 0.05 

**so we will reject the Null hypothesis i.e. there is raw cornstarch effect.**


Problem Statement 2:

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.

What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?

In [3]:
n1 = 100
n2 = 100
R1 = 0.52            #Republicans from state 1
D1 = 0.48            #Democrats from state 1
R2 = 0.47            #Republicans from state 2
D2 = 0.53            #Democrats from state 2

mu = R1 - R2
print(f"mu: {mu}")
std = math.sqrt(((R1 * D1 ) / n1) + ((R2 * D2) /n2))
print(f"std: {std}")

#This problem requires us to find the probability that p1 is less than p2. 
#This is equivalent to finding the probability that R1 - R2 < 0. 
x = 0
#To find this probability, we need to transform the random variable (R1 - R2) into a z-score. 
#That transformation appears below.

Z_R1_R2 = ( x - mu)/std
print(f"Z_p1_p2 : {Z_R1_R2}")

#From Z table we find that the probability of a z-score being -0.7082 or less is 0.24.

#Therefore, the probability that the survey will show a greater percentage of Republican voters 
#in the second state than in the first state is 0.24

mu: 0.050000000000000044
std: 0.07061869440877536
Z_p1_p2 : -0.7080278164104213


Problem Statement 3:

You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?

In [4]:
#The z score tells you how many standard deviations from the mean your score is
x = 1100 #
mu = 1026 # Population Mean
sd = 209 #population standard deviation
z = ( x - mu)/sd
print("Z Score : ",z)
#the above calculation shows that my score is 0.35 standard deviations above the mean
print("My Score is in the range {} - {}  with a  zscore {:.2f}".format(mu - sd,mu + sd,z))

Z Score :  0.35406698564593303
My Score is in the range 817 - 1235  with a  zscore 0.35
